src/altinn-app-frontend/src/features/form/layout/index.ts

```diff
@@ -99,6 +99,17 @@ export interface ILayoutCompDatePicker extends ILayoutCompBase<'DatePicker'> {
 }
 
 export type ILayoutCompDropdown = ILayoutCompBase<'Dropdown'> & ISelectionComponent;
+export interface ILayoutCompList extends ILayoutCompBase<'List'> {
+  tableHeaders?: string[];
+  sortableColumns?: string[];
+  pagination: IPagination;
+  dataList?: IDataList[];
+  dataListId?: string;
+  mapping?: IMapping;
+  secure?: boolean;
+  source?: IOptionSource;
```

Comment:
Could you add the component to `schemas/json/layout/layout.schema.v1.json` as well? :pray: In JsonSchema it's also possible to require either `dataList` OR `dataListId`